Doc/using/cmdline.rst

```diff
@@ -994,3 +994,101 @@ Debug-mode variables
+directory with the platform-independent Python modules is called ``prefix``.
+The directory with the shared libraries is called ``exec_prefix``.
+
+The :envvar:`PYTHONHOME` environment variable may be used to set the ``prefix``
+and ``exec_prefix`` locations. Otherwise these directories are found by using
+the Python executable as a starting point and then looking for various 'landmark'
+files and directories. Note that any symbolic links are followed so the real
+Python executable location is used as the search starting point. On MacOS, the
+:envvar:`PYTHONEXECUTABLE` environment variable may be used to set the Python
+executable location. The Python executable location is called ``home``.
+
+Once ``home`` is determined, the ``prefix`` directory is found by first looking
+for :file:`python{majorversion}{minorversion}.zip` (``python311.zip``). On Windows
+the zip archive is searched for in ``home`` and on Unix the archive is expected
+to be in :file:`lib`. Note that the expected zip archive location is added to the
+module search path even if the archive does not exist. If no archive was found,
+Python on Windows will continue the search for ``prefix`` by looking for :file:`Lib\\os.py`
+or :file:`Lib\\os.pyc`. Python on Unix will look for :file:`lib/python{majorversion}.{minorversion}/os.py`
+(``lib/python3.11/os.py``) or :file:`lib/python{majorversion}.{minorversion}/os.pyc` (``lib/python3.11/os.pyc``).
+On Windows ``prefix`` and ``exec_prefix`` are the same, however on other platforms
+:file:`lib/python{majorversion}.{minorversion}/lib-dynload` (``lib/python3.11/lib-dynload``)
+is searched for and used as an anchor for ``exec_prefix``.
+
+Once found, ``prefix`` and ``exec_prefix`` may be accessed at :data:`sys.prefix`
+and :data:`sys.exec_prefix`.
+
+Finally, the :mod:`site` module is processed and :file:`site-packages` directories
+are added to the module search path. A common way to customize the search path is
+to create :mod:`sitecustomize` or :mod:`usercustomize` modules as described in
+the :mod:`site` module documentation.
+
+.. note::
+
+   The effect of :mod:`site` on the module search path may be seen by running
+   Python with ``-S`` which will start Python without importing site.
+
+Virtual environments
+~~~~~~~~~~~~~~~~~~~~
+
+If Python is run in a virtual environment (as described at :ref:`tut-venv`)
+then ``prefix`` and ``exec_prefix`` are specific to the virtual environment.
+
+If a ``pyvenv.cfg`` file is found alongside the main executable, or in the
+directory one level above the executable, the following variations apply:
+
+* If ``home`` is an absolute path and :envvar:`PYTHONHOME` is not set, this
+  path is used instead of the path to the main executable when deducing ``prefix``
+  and ``exec_prefix``.
+
+_pth files
+~~~~~~~~~~
+
+To completely override :data:`sys.path`, create a ``._pth`` file with the same
+name as the DLL/dylib (``python311._pth``) or the executable (``python._pth``)
```

Comment:
.dylib is Apple (macOS/iOS) specific, .so is the used on most other Unix-systems. Perhaps we should just use the generic-term `shared library` throughout this section and explain once about the various platform-specific terms .so., DLL, and .dylib. Many users will have no idea what these other prefixes mean on their platforms.

Reply:
Thank you, I have amended as suggested.